apps/dashboard/src/app/resources/edit/add-field/add-field.component.html

```diff
@@ -0,0 +1,87 @@
+<h5>{{ 'resource.field.add' | translate }}</h5>
+
+<nsi-dropdown-button [popupRef]="fieldTypeDropdown">
+  {{ 'resource.field-' + fieldType.value | translate }}
+</nsi-dropdown-button>
+<pa-dropdown #fieldTypeDropdown>
+  <pa-option
+    *ngFor="let type of availableTypes"
+    [value]="type"
+    (selectOption)="changeFieldType(type)">
+    {{ 'resource.field-' + type | translate }}
+  </pa-option>
+</pa-dropdown>
+
+<form [formGroup]="form">
+  <ng-container *ngIf="fieldType.value === 'text'">
+    <div class="field-block">
+      <label
+        class="title-xxs"
+        for="format">
+        {{ 'resource.field-text-format-label' | translate }}
+      </label>
+      <pa-select
+        id="format"
+        formControlName="format"
+        externalLabel>
+        <pa-option
+          *ngFor="let format of availableFormats"
+          [value]="format">
+          {{ format }}
+        </pa-option>
+      </pa-select>
+    </div>
+
+    <div class="field-block">
+      <label
+        class="title-xxs"
+        for="text">
+        {{ 'resource.field-text-body-label' | translate }}
+      </label>
+      <pa-textarea
+        id="text"
+        formControlName="text"
+        externalLabel></pa-textarea>
+    </div>
+  </ng-container>
+
+  <ng-container *ngIf="fieldType.value === 'link'">
+    <div class="field-block">
+      <label
+        class="title-xxs"
+        for="link">
+        URL
+      </label>
+      <pa-input
```

Comment:
As far as I remember, if the link URL doesn't include the URL schema (http/https), the backend can't process the link. In the link upload modal this issue is validated. Not sure if it should be validated here.

Reply:
Good point, I didn't think about that. I'll add a validator